sockjs 0.4

pahaz opened the pull request:
Hi. I finde some problems with the lose event loop on my tests. 

Warnings:
```
\venv\lib\site-packages\aiohttp\web_ws.py:80: DeprecationWarning: use .prepare(request) instead
  warnings.warn('use .prepare(request) instead', DeprecationWarning)
venv\lib\site-packages\aiohttp\web_reqrep.py:675: DeprecationWarning: use .prepare(request) instead
  warnings.warn('use .prepare(request) instead', DeprecationWarning)
Python35\lib\asyncio\tasks.py:522: DeprecationWarning: asyncio.async() function is deprecated, use ensure_future()
  DeprecationWarning)
```

Now I have only one warning:
```
<CoroWrapper SessionManager.clear() running at venv\lib\site-packages\sockjs\session.py:388, created at venv\lib\site-packages\sockjs\session.py:406> was never yielded from
Coroutine object created at (most recent call last):
  File "venv\lib\site-packages\sockjs\session.py", line 406, in __del__
    self.clear()
```

I think this is correct.

jettify commented on sockjs/transports/websocket.py:
```diff
@@ -76,11 +76,14 @@ def process(self):
 
-            server = asyncio.async(self.server(ws, self.session))
-            client = asyncio.async(self.client(ws, self.session))
+            server = asyncio.ensure_future(
```
This change is backward incompatible, you need something like: https://github.com/KeepSafe/aiohttp/blob/master/aiohttp%2Fhelpers.py#L17-L20

pahaz replied:
Con you provide any info about sported aiohttp version? I will fix it tomorrow.

fafhrd91 replied:
@pahaz could you replace `asyncio.ensure_future` with `aiohttp.helpers.ensure_future`

pahaz commented:
@jettify, can you tall something about failed build?

jettify commented:
I suspect that method `make_request` is not longer valid because of changes in aiohttp. Basically it creates mocked version of request object and one more attribute should be patched or so.

pahaz commented:
Yes, but I my question more about your plane to fix it. Do you have this plane? And what do you think about fix aiohttp version? (this is not a first similar problem)

pahaz commented:
The problem in this commit: https://github.com/KeepSafe/aiohttp/commit/fa06acc2392c516491bdb25301ad3ef2b700ff5f

pahaz commented:
Fixed! Can you merge this pull and update pypi version?

fafhrd91 commented:
@pahaz would you do this yourself? i can add you to comitters

pahaz commented:
@fafhrd91, great, now I do some project based on `sockjs` this would be helpful. My PYPI username is the same.

fafhrd91 commented:
@asvetlov objections?

fafhrd91 commented:
@pahaz done
please run sockjs test suit before release

pahaz commented:
@fafhrd91, ok. This is a mandatory rule.

pahaz commented:
Is the backward compatible required? This patch incompatible with `0.15.1` only with https://github.com/KeepSafe/aiohttp/blob/master/HISTORY.rst#0180-19-10-2015

Problem in this changes (https://github.com/aio-libs/sockjs/pull/19/files#diff-b68833d56503644b348390fc48b2f56bR67)
```
Deprecate StreamResponse.start() method in favor of StreamResponse.prepare() coroutine #525

.start() is still supported but responses begun with .start() doesn't call signal for response preparing to be sent.
```

Add backward compatibility hack or up aiohttp version? What preferably?

fafhrd91 commented:
i think current stable release of `aiohttp` is enough. maybe previous, but that is more than enough

fafhrd91 commented:
regarding asyncio and python, sockjs should be compatible with same version of python as aiohttp.

pahaz commented:
Now we have problem with aiohttp 0.20.2 (surrent stable). I opened issues https://github.com/KeepSafe/aiohttp/issues/760 and https://github.com/aio-libs/sockjs/issues/20

pahaz commented:
Now its work with latest stable aiohttp (0.20.2) and aiohttp master.

pahaz commented:
I`m ready to merge and update PYPI version. Are there any objections? 

Changes:
- test refactoring + remove code duplication
- fixed pep8 syle in `sockjs`
- fixed deprection warnings
- add CI pep8 checks
- CI tests run on aiohttp==0.18.3, aiohttp latest stable and aiohttp master

I planned for the end of the day release a new version.

asvetlov commented:
LGTM but @pahaz please wait at least a day after asking for objections before the merge :)

Also `pip uninstall aiohttp -y && pip install aiohttp==0.18.3` looks ugly -- it should be done either by tox tool or by travis build matrix.

pahaz commented:
@asvetlov, Indeed, the next time will be more patient :)

asvetlov commented:
@pahaz please add git tag and github release notes next time.

pahaz commented:
Yes. But someone already add tags too (`0.4` and `v0.4`). I deleted unused
tags and add `0.4v` as continuation of `0.2v`, `0.3v` notation.

On Thu, Feb 4, 2016 at 4:09 PM, Andrew Svetlov <notifications@github.com>
wrote:

> @pahaz <https://github.com/pahaz> please add git tag and github release
> notes next time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aio-libs/sockjs/pull/19#issuecomment-179774395>.
>

asvetlov commented:
I did add `v0.4` tag.
Actually I don't like `0.4v` -- it's named against schema used in git docs etc.
Moreover in all other aio-libs we use `v0.4.0` tag naming.
